src/modules/DASDController.rb

```diff
@@ -274,6 +274,13 @@ def GetDevices
       deep_copy(@devices)
     end
 
+    # @param channel [String] "0.0.0000" "ab.c.Def0"
+    # @return [String] "0000000" "abcdef0"
+    def channel_sort_key(channel)
+      parts = channel.downcase.split(".", 3)
+      format("%02s%1s%4s", parts[0], parts[1], parts[2])
```

Comment:
I could be wrong but I think this is not enough because the libyui sort relies on stoll which will return '0' with any string if it is not able to convert to a integer, that is why I commented to convert it using `to_hex.to_i` but feel free to correct me :P

Reply:
I tested it at the last moment and you were right